pyinteraph/path_analysis.py

```diff
@@ -23,76 +23,85 @@ def build_graph(fname, pdb = None):
-    except Exception:
-        raise ValueError(f"Residue not in PDB or incorrect format: {res}")
-    colon_replace = []
-    # Substitute range of residues with the actual residues
-    for inp in input_colon:
-        try:
-            # Create list of size two with start and end of range
-            colon_split = inp.split(':')
-            # Find the index of those res in the indentifiers list
-            index = [identifiers.index(res) for res in colon_split]
-            # Replace with the residues in that range
-            inp_replace = identifiers[index[0]:index[1]+1]
-            # Concatenate to list
-            colon_replace += inp_replace
-        except Exception:
-            # Report if the specified range does not exist in the PDB
-            raise ValueError(f"Residue range not in PDB or incorrect format: {inp}")
-    # Add both lists
-    input_list = comma_list + colon_replace
+    res_list = []
+    # Split by comma and then colon to make list of list
+    split_list = [elem.split(':') for elem in user_input.split(',')]
+    for sub_list in split_list:
+        # If list size is one, word is separated by comma
+        if len(sub_list) == 1:
+            try:
+                res = sub_list[0]
+                # Find index
+                identifiers.index(res)
+                # Append to list
+                res_list.append(res)
+            except:
+                raise ValueError(f"Residue not in PDB: {res}")
+        # If list size is 2, word is separated by a single colon
+        elif len(sub_list) == 2:
+            try:
+                # Get indexes for residue in identifiers
+                index = [identifiers.index(res) for res in sub_list]
+                # Create list with all residues in that range
+                res_range = identifiers[index[0]:index[1]+1]
+                # Concatenate with output list
+                res_list += res_range
+            except:
+                raise ValueError(f"Residue range not in PDB: {':'.join(sub_list)}")
+        # Other list sizes means multiple colons
+        else:
+            err_str = f"Incorrect format, only one ':' allowed: {':'.join(sub_list)}"
+            raise ValueError(err_str)
     # Remove duplicates
-    input_list = list(set(input_list))
-    return input_list
+    res_list = list(set(res_list))
+    return res_list
+
+
+def get_persistence_graph(graph, paths, identifiers):
+    """Takes in a PSN graph, a list of paths and a list of nodes in the 
+    PSN and returns a new graph of paths where each edge corresponds to 
+    persistence value of the original PSN. All nodes from the PSN are
+    also included in the output graph.
+    """
 
-def get_shortest_paths(graph, source, target, maxl):
+    pers_graph = nx.Graph()
+    # Add all paths
+    for p in paths:
+        pers_graph.add_path(p)
+    # Add all edge weights
+    for u, v in pers_graph.edges():
+        pers_graph[u][v]['weight'] = graph[u][v]['weight']
```

Comment:
Is there no way in networkx to assign the value of an edge attribute to multiple edges simultaneously? It could be faster than a for loop.